snap/info_snap_yaml.go

```diff
@@ -319,6 +319,10 @@ func setComponentsFromSnapYaml(y snapYaml, snap *Info, strk *scopedTracker) erro
 	}
 
 	for name, data := range y.Components {
+		if data.Type == KernelModulesComponent && snap.Type() != TypeKernel {
```

Comment:
nitpick: should we implement this using a map of optional component type => snap types? even if for a long time there is only one entry it might jump more into readers' eyes?

Reply:
I've done this now